File: src/main/java/com/baulsupp/oksocial/services/google/DiscoveryParameter.java
package com.baulsupp.oksocial.services.google;

import java.util.Map;

public class DiscoveryParameter {
  private String name;
  private final Map<String, Object> map;

  public DiscoveryParameter(String name, Map<String, Object> map) {
    this.name = name;
    this.map = map;

  }

  public String name() {
    return name;
  }

  public String type() {
    return (String) map.get("type");
  }

  public String description() {
    return (String) map.get("description");
  }

  public String location() {
    return (String) map.get("location");
  }

  public String pattern() {
    return (String) map.get("pattern");
  }
}

File: src/main/kotlin/com/baulsupp/oksocial/services/google/DiscoveryParameter.kt
package com.baulsupp.oksocial.services.google

import com.baulsupp.oksocial.services.google.model.Parameter

class DiscoveryParameter(private val name: String, private val parameter: Parameter) {
  fun name(): String = name

  fun type(): String? = parameter.type

  fun description(): String? = parameter.description

  fun location(): String? = parameter.location

  fun pattern(): String? = parameter.pattern
}
